torch/_dynamo/symbolic_convert.py

```diff
@@ -1831,6 +1833,13 @@ def __init__(
         vars = list(code_options["co_varnames"])
         vars.extend(x for x in self.cell_and_freevars() if x not in vars)
 
+        if config.dynamic_shapes and dynamic_plan:
+            for k in vars:
+                if k in dynamic_plan:
+                    tensor_at_k = f_locals[k]
+                    assert isinstance(tensor_at_k, torch.Tensor)
+                    torch._dynamo.mark_dynamic(tensor_at_k, dynamic_plan[k])
```

Comment:
Don't use mark_dynamic. Instead you want to control the constraints/policy we send to ShapeEnv.

Reply:
But this just does that for us? why not use mark_dynamic?

Reply:
One problem with using `mark_dynamic` is that you will get `RelaxedUnspecConstraint` which will fail compilation if we end up guarding on this dim. But this is not desirable behavior, as you've observed earlier: you don't want to error if we specialize a dimension we *automatically* inferred as dynamic. If you use `mark_dynamic` there's no way to thread the needle and have a different type of constraint for these.

Reply:
Yes, that is true. One way would be to ban mixing automatic and mark, but that feels lame. 

Do you want me to try to take a stab at a different kind of constraint here, with a different mark API?